website/docs/flow/inspector.md

```diff
@@ -1,38 +1,63 @@
 ---
-title: Inspector
+title: Flow Inspector
 ---
 
-The flow inspector, introduced in 2021.10, allows administrators to easily figure out how custom flows work, inspect the current context and debug issues.
+The flow inspector, introduced in 2021.10, allows administrators to visually determine how custom flows work, inspect the current [flow context](../flow/context/index.md), and investigate issues.
+
+![](./flow-inspector.png)
 
 :::info
-When running a flow with the inspector enabled, the flow is still executed normally. This means that for example, a [User write](../flow/stages/user_write.md) stage will _actually_ write user data.
+Be aware that when running a flow with the inspector enabled, the flow is still executed normally. This means that for example, a [User write](../flow/stages/user_write.md) stage will _actually_ write user data.
 :::
 
-### Accessing the inspector
+## Access the Flow Inspector
+
+### Permissions and debug mode
+
+By default, the inspector is only enabled when the currently authenticated user is a superuser, OR if a user has been granted the [permission](../user-group-role/access-control/permissions.md) **Can inspect a Flow's execution** (or is a user assigned to role with the permission).
+
+When developing authentik with the debug mode enabled, the inspector is enabled by default and can be accessed by both unauthenticated users and standard users. However the debug mode should only be used for the development of authentik. So unless you are a developer and need the more verbose error information, the best practice for using the flow inspector is to assign the permission, not use debug mode.
+
+### Open the Flow Inspector
+
+1. To access the inspector, open the Admin interface and navigate to **Flows and Stages -> Flows**.
+
+2. Select the specific flow that you want to inspect by clicking its name in the list.
 
-By default, the inspector can only be enabled when the currently authenticated user is a superuser.
-When developing authentik with the debug-mode enabled, the inspector is enabled by default and can be accessed by both unauthenticated users and standard users. However the debug-mode should only be used for the development of authentik.
+3. On the Flow's detail page, on the left side under **Execute Flow**, click **with inspector**.
+
+4. The selected flow will launch in a new browser window, with the Inspector displayed to the right.
+
+Alternatively, to launch the inspector a user with the correct permission can add the query parameter `?inspector` to the URL when a flow is designated in the URL; that is when you have a flow open.
```

Comment:
Suggestion: "Alternatively, a user with the correct permission can launch the inspector by adding the query parameter `?inspector` to the URL when the URL opens on a flow."